### ISSUE TITLE
Job promise is an ECMAScript type, not WebIDL

### ISSUE THREAD

**Issue by ADKaster** [w3c/ServiceWorker#1731]
Is there any meaningful reason that the job promise (https://w3c.github.io/ServiceWorker/#dfn-job-promise) is linked to the ECMAScript Promise definition (https://tc39.es/ecma262/#sec-promise-objects) rather than WebIDL Promise, AKA ECMAScript PromiseCapability (https://webidl.spec.whatwg.org/#idl-promise) ?

I noticed that none of the Job promise-related AOs reference the WebIDL promise AOs, and just say to "resolve" or "reject" the promise.

I would expect that the spec would call out to the resolve (https://webidl.spec.whatwg.org/#resolve), reject (https://webidl.spec.whatwg.org/#reject) and react (https://webidl.spec.whatwg.org/#dfn-perform-steps-once-promise-is-settled) WebIDL operations as necessary.

**Comment by mkruisselbrink** [comment]
I'd say the main reason is that most of this spec predates webidl having all those operations, and nobody ever got around to updating the spec when webidl started including better support for promise related operations...

**Comment by ADKaster** [comment]
Makes sense, thanks! I'll see if I can wrangle up a PR in the coming days then.
